sdk/core/azure-core/src/http/logging_policy.cpp

```diff
@@ -30,22 +30,69 @@ std::string TruncateIfLengthy(std::string const& s)
   return s.substr(0, BeginLength) + Ellipsis + s.substr(length - EndLength, EndLength);
 }
 
-std::string GetRequestLogMessage(Request const& request)
+std::string GetRequestLogMessage(
+    Azure::Core::Http::LoggingPolicyOptions const& options,
+    Request const& request)
 {
   std::ostringstream log;
 
-  log << "HTTP Request : " << HttpMethodToString(request.GetMethod()) << " "
-      << request.GetUrl().GetAbsoluteUrl();
+  std::string url = request.GetUrl().GetAbsoluteUrl();
+  if (!url.empty())
```

Comment:
consider returning if url is empty.  We won't find ? or anything else below if the AbsoluteUrl is empty.

```cpp
if (url.empty()) {
  return {}; // empty string or any placeholder you want here.
}
```

This might be, however, a very unlikely scenario to reach the logging policy with an empty Url.  I would expect that to be thrown much earlier from sdk clients

Reply:
I don't think any special logic needs to be there. If the URL is empty, it should be caught by clients. Logger's job is to log everything as it sees it, so that diagnostics can be done. Plus, if at some point empty URL has some meaning (useful in oversimplified unit tests?), you don't need to update this place. I only need logging to not blow up if URL is empty, and then if it is, we log it as is, without thinking out of our scope what this might mean. I think it is reasonable.

Reply:
> I don't think any special logic needs to be there

That's what I meant. We don't need to check if url is empty() here.
We don't need to manually parse the url to check if it contains queryParams or not and brake it into url without query.  Just update our Url class to return what you need directly.  The Url knows if there are query params or not already.